test/ILLink.RoslynAnalyzer.Tests/TestChecker.cs

```diff
@@ -56,16 +57,35 @@ public void ValidateAttributes (List<AttributeSyntax> attributes)
 			}
 		}
 
+		private bool IsProducedByAnalyzer (Dictionary<string, ExpressionSyntax> args)
+		{
+			return !args.TryGetValue ("ProducedBy", out var diagnosticProducedBy) || (
```

Comment:
Could we do something similar to what we have in `IsProducedByLinker` instead of having this very long expression?